nav2_behavior_tree/include/nav2_behavior_tree/plugins/control/round_robin_node.hpp

```diff
@@ -42,8 +43,9 @@ namespace nav2_behavior_tree
  * | FAILURE | RUNNING |  IDLE   |  - A returns FAILURE so B gets ticked and returns RUNNING
  *                                  - RoundRobin returns RUNNING and C is not ticked yet
  * | FAILURE | SUCCESS |  IDLE   |  - B returns SUCCESS, so RoundRobin halts all children and
- *                                  - returns SUCCESS, next iteration will tick C.
- * | RUNNING |  IDLE   | FAILURE |  - C returns FAILURE, so RoundRobin circles and ticks A.
+ *                                  - returns SUCCESS, next iteration will tick B again.
```

Comment:
I'm not sure that this change in behavior is really what we want. For instance, in `navigate_w_replanning_and_round_robin_recovery.xml` in this block
```
<RoundRobin name="GlobalPlannerRecoveryActions">
     <ClearEntireCostmap name="ClearGlobalCostmap-Context" service_name="global_costmap/clear_entirely_global_costmap"/>
    <Wait wait_duration="2"/>
</RoundRobin>
```
If `clearcostmap` fails, we'd advance to `wait` on the next tick. `wait` seems unlikely to ever fail, so we'd be stuck on that recovery action forever

Reply:
You're right, going to revert this